src/html/mod.rs

```diff
@@ -1145,6 +1145,190 @@ pub fn generate(
   Ok(files)
 }
 
+pub fn generate_json(
```

Comment:
this blob has a decent amount of duplicated logic. it is not often-changing, but still. i'll think about how this could be deduplicated in a follow-up